### PR TITLE
Update solutions of some tests

### LINKED ISSUE #9: KNITRO finds different solutions for some tests
I am currently testing the MOI branch of KNITRO.jl on [od/moi](https://github.com/lanl-ansi/MINLPTests.jl/tree/od/moi). The package is working well, and almost all tests pass! 

However, using the following Optimizer factory: 
```julia
solver = JuMP.with_optimizer(KNITRO.Optimizer, opttol=1e-10)
```
I get different results for some tests: 
- [nlp_005_010](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp/005_010.jl) (`1.5449723904763646` instead of `1.5449760741521967`)
- [nlp_mi_003_014](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp-mi/003_014.jl) and [nlp_mi_003_015](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp-mi/003_015.jl) (`11.0` instead of `11.000000198181866`)
- [nlp_mi_004_010](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp-mi/004_010.jl) and [nlp_mi_004_011](https://github.com/lanl-ansi/MINLPTests.jl/blob/od/moi/src/nlp-mi/004_011.jl) (`-4.5768807316995614` instead of ` -4.576881309190901`)

What guarantee do we have on the value of the solutions? Is it worth to consider more than 8 digits for MIP problems?

### PR DESCRIPTION
**Pull request by frapac**
Knitro seems to give better results on a small subset of the NLP tests. I have updated the solutions with the results found by Knitro. 

This sould resolve #9

### CODE REVIEW

File: src/nlp-mi/004_010.jl [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
     @constraint(m, -1.2*x - y <= z/1.35)
-    
+
     optimize!(m)
-    
+
     check_status(m, termination_target, primal_target)
-    check_objective(m, -4.576881309190901, tol = objective_tol)
+    check_objective(m, -4.5768807317, tol = objective_tol)
     check_solution([x,y,z], [-0.9969558935432884, -0.077968812655046, 3], tol = primal_tol)
```
Comment (odow) [inline src/nlp-mi/004_010.jl]: Do you need to change the solution as well? At the moment they don't line up.

**Comment by odow** [comment]
You should also relax the default tolerances so that Ipopt can pass the tests by default.

**Comment by ccoffrin** [comment]
Let me review this in a bit.  Ideally all of them problems in this repo
should be trivial for any solver.  If knitro finds a better solution, I
would prefer to change the problem so that both find the same solution.

**Comment by odow** [comment]
I think this is just a case of using different termination tolerances in the solvers, and having functions like `tan(x)`. If Ipopt uses a default tolerance of `1e-6`, we can't expect it to get 10 digits of precision...

**Comment by frapac** [comment]
IMO, we should keep the tests, even if the solvers give different results on them. That could help to identify basic problems.

**Comment by ccoffrin** [comment]
All of these changes look fine to me.  We just need to tweak the check tolerances so that these pass with Ipopt / Juniper.

**Comment by frapac** [comment]
Ok, I am updating the PR. Thanks for the feedbacks!

File: src/nlp-mi/004_010.jl (hunk 1 above)
Comment (frapac) [reply src/nlp-mi/004_010.jl]: The test passes for the solution with a tolerance set at 1e-7 and Knitro as a solver.

**Pull request merged**